folium/features.py

```diff
@@ -241,6 +241,14 @@ def __init__(self, data, style_function=None):
             else:  # This is a filename
                 self.embed = False
                 self.data = data
+        elif hasattr(data, 'to_crs'):
```

Comment:
Let's use `isinstance`  I have a few custom object with `to_crs()` defined :wink: 
Also, if we use `geopandas.geodataframe.GeoDataFrame` we can probably drop the next `if` below.